polaris.shopify.com/pages/examples/index-table-with-nested-rows.tsx

```diff
@@ -0,0 +1,233 @@
+    },
+    {
+      id: '2562',
+      quantity: 30,
+      price: '$975',
+      size: 'Medium Medium Medium',
+      color: 'Orange Orange Orange',
+    },
+    {
+      id: '4102',
+      quantity: 27,
+      price: '$2885',
+      size: 'Large',
+      color: 'Orange',
+    },
+    {
+      id: '2564',
+      quantity: 19,
+      price: '$1,209',
+      size: 'Small',
+      color: 'Red',
+      disabled: true,
+    },
+    {
+      id: '2563',
+      quantity: 22,
+      price: '$1,400',
+      size: 'Small',
+      color: 'Green',
+    },
+  ];
+
+  const columnHeadings = [
+    {title: 'Name', id: 'column-header--size'},
+    {
+      hidden: false,
+      id: 'column-header--price',
+      title: 'Price',
+    },
+    {
+      alignment: 'end',
+      id: 'column-header--quantity',
+      title: 'Available',
+    },
+  ];
+
+  const groupRowsByGroupKey = (
+    groupKey: keyof Product,
+    resolveId: (groupVal: string) => string,
+  ) => {
+    let position = -1;
+    const groups: Groups = rows.reduce((groups: Groups, product: Product) => {
+      const groupVal: string = product[groupKey] as string;
+      if (!groups[groupVal]) {
+        position += 1;
+
+        groups[groupVal] = {
+          position,
+          products: [],
+          id: resolveId(groupVal),
+        };
+      }
+      groups[groupVal].products.push({
+        ...product,
+        position: position + 1,
+      });
+
+      position += 1;
+      return groups;
+    }, {});
+
+    return groups;
+  };
+
+  const resourceName = {
+    singular: 'product',
+    plural: 'products',
+  };
+
+  const {selectedResources, allResourcesSelected, handleSelectionChange} =
+    useIndexResourceState(rows as unknown as {[key: string]: unknown}[], {
+      resourceFilter: ({disabled}) => !disabled,
+    });
+
+  const groupedProducts = groupRowsByGroupKey(
+    'color',
+    (color) => `color--${color.toLowerCase()}`,
+  );
+
+  const rowMarkup = Object.keys(groupedProducts).map((color, index) => {
+    const {products, position, id: subheaderId} = groupedProducts[color];
+    let selected: IndexTableRowProps['selected'] = false;
+
+    const someProductsSelected = products.some(({id}) =>
+      selectedResources.includes(id),
+    );
+
+    const allProductsSelected = products.every(({id}) =>
+      selectedResources.includes(id),
+    );
+
+    if (allProductsSelected) {
+      selected = true;
+    } else if (someProductsSelected) {
+      selected = 'indeterminate';
+    }
+
+    const selectableRows = rows.filter(({disabled}) => !disabled);
+    const rowRange: IndexTableRowProps['selectionRange'] = [
+      selectableRows.findIndex((row) => row.id === products[0].id),
+      selectableRows.findIndex(
+        (row) => row.id === products[products.length - 1].id,
+      ),
+    ];
+
+    const disabled = products.every(({disabled}) => disabled);
+
+    return (
+      <Fragment key={subheaderId}>
+        <IndexTable.Row
+          rowType="subheader"
```

Comment:
If this is an example of `nested rows` with `range selection` - it should not contain a `subheader` design treatment as we already have that in the [docs](https://polaris.shopify.com/components/tables/index-table?example=index-table-with-subheaders).

```suggestion
          rowType="parent"
```

If the `parent` type doesn't exist and/or there is no need for it to exist - we can default to `data` instead of `subheader`.